File: src/foam/core/Agency.js
/**
 * @license
 * Copyright 2019 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */
foam.INTERFACE({
  package: 'foam.core',
  name: 'Agency',

  methods: [
    {
      name: 'submit',
      type: 'Void',
      args: [
        { name: 'x', type: 'Context' },
        { name: 'agent', type: 'foam.core.ContextAgent' },
        { name: 'description', type: 'String' }
      ]
    }
  ]
});

File: src/foam/core/Agency.java
/**
 * @license
 * Copyright 2019 The FOAM Authors. All Rights Reserved.
 *   http://www.apache.org/licenses/LICENSE-2.0
 */
package foam.core;

public interface Agency {
  public void submit(foam.core.X x, foam.core.ContextAgent agent);
  public void submit(foam.core.X x, foam.core.ContextAgent agent, String description);
}